Building a TypeScript application fails due to new declarations output

1. Create a new application with TypeScript (with pnpm and React in my case)
2. Execute a build

Resulting in this error:
[build:types] tsconfig.json(5,5): error TS5069: Option 'declarationDir' cannot be specified without specifying option 'declaration' or option 'composite'.

--- a/.changeset/young-grapes-call.md
+++ b/.changeset/young-grapes-call.md
@@ -1,0 +1,5 @@
+---
+"generator-single-spa": patch
+---
+
+Upgrade to ts-config-single-spa@3

--- a/packages/generator-single-spa/src/common-templates/typescript/typescript.package.json
+++ b/packages/generator-single-spa/src/common-templates/typescript/typescript.package.json
@@ -7,7 +7,7 @@
     "eslint-config-ts-important-stuff": "^1.1.0",
     "typescript": "^4.3.4",
     "webpack-config-single-spa-ts": "^3.0.0",
-    "ts-config-single-spa": "^2.0.1"
+    "ts-config-single-spa": "^3.0.0"
   },
   "dependencies": {
     "@types/jest": "^26.0.23",

